### PR TITLE
Add Slack webhook support.

### PR DESCRIPTION
**Pull request by bmenasha**
Defines the new `violations_slack_webhook` variable which will configure a slack
webhook for all notifiers. The expectation is that per notifier slack hooks are
unlikely to be useful. The existing `iam_policy_violations_slack_webhook` and `kms_violations_slack_webhook` still work and
take precedence over this new variable if both are defined.

### CODE REVIEW

File: test/integration/simple_example/controls/server.rb [hunk 1]
```diff
@@ -324,8 +324,6 @@
             including(
               "resource" => "iam_policy_violations",
               "should_notify" => true,
-              # This extra bit of verification tests the `iam_policy_violations_slack_webhook` variable
```
Comment (morgante) [inline test/integration/simple_example/controls/server.rb]: @bmenasha Perhaps we could repurpose this test? I'm fine with dropping it though.

File: modules/server/templates/configs/forseti_conf_server.yaml.tpl [hunk 2]
```diff
@@ -301,6 +303,12 @@ notifier:
                 data_format: csv
                 # gcs_path should begin with "gs://"
                 gcs_path: gs://${FORSETI_BUCKET}/scanner_violations
+            %{ if VIOLATIONS_SLACK_WEBHOOK != "" }
```
Comment (ocervell) [inline modules/server/templates/configs/forseti_conf_server.yaml.tpl]: Should we add a `AUDIT_LOGGING_VIOLATIONS_SLACK_WEBHOOK` variable like on line 288 ? Same for all other webhooks

File: test/integration/simple_example/controls/server.rb (hunk 1 above)
Comment (bmenasha) [reply test/integration/simple_example/controls/server.rb]: I'm thinking about moving the whole template into it's own sub-module in the hopes of writing more tests against it. but i don't want to do that in this pull request.

File: modules/server/templates/configs/forseti_conf_server.yaml.tpl (hunk 2 above)
Comment (bmenasha) [reply modules/server/templates/configs/forseti_conf_server.yaml.tpl]: i was hoping to void webhooks for each resource type. seems like an excessive # of variables and i don't see the use case for having resource specific webhooks.

File: test/integration/simple_example/controls/server.rb (hunk 1 above)
Comment (morgante) [reply test/integration/simple_example/controls/server.rb]: Got it, feel free to follow up.

**Comment by bmenasha** [comment]
> The example needs to enable a Slack webhook or the tests need to no longer check for its presence.

I thought I did the latter in this change? is it failing?
thanks

**Comment by morgante** [comment]
Yes, it looks like tests are failing: https://concourse.infra.cft.tips/builds/4294

**Comment by bmenasha** [comment]
thanks i see that i missed one. fixed.

**Comment by dekuhn** [comment]
CI/CD is passing now did the test needing to get fix get resolved?

**Comment by aaron-lane** [comment]
@dekuhn yes, it was fixed by https://github.com/forseti-security/terraform-google-forseti/pull/168/commits/98e8b9f8455c6da23e0a1e64e278aa8bea4dddf9.

**Pull request merged**